ovn-fake-multinode-utils/playbooks/install-dependencies.yml

```diff
@@ -2,6 +2,24 @@
   hosts: ovn_hosts
   tasks:
     - name: Install required packages
-      shell: yum install git gcc openvswitch python3-pyyaml python3-virtualenv python3-devel containers-common --skip-broken -y
+      shell: |
+        yum install git gcc openvswitch python3-pyyaml python3-virtualenv \
+            python3-devel containers-common --skip-broken -y
```

Comment:
Can we use `ansible.builtin.package` module for this as well?
`ansible.builtin.dnf/yum` should also be fine, since we're in the OS-specific part here.

Reply:
The reason for not doing that was the `--skip-broken` part in the original code.

If the reason for doing that is now gone, or we can accomplish something similar with Ansible, I'm fine with changing it. Anyone know why it was there in the first place?

Reply:
I'm not sure why it is there.  But `ansible.builtin.dnf` module has it: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/dnf_module.html#parameter-skip_broken

Reply:
Thx for the pointer!

After some light inspection, it does appear Ansible would potentially pass on task variables to whatever backend package module is used, ref: https://github.com/ansible/ansible/blob/bad8843124a50493141a3e3d7920353239021389/lib/ansible/plugins/action/package.py

I do not know for a fact that it works that way though. I'll try to add the `ansible.builtin.package` module and we'll see if that is inherited to the `dnf` module on RHEL systems?

I don't have one at hand so would appreciate if one of you would be able to check whether it works there?

Reply:
Done

Reply:
I think, you missed the `name:` part.

Reply:
Nah, that didn't work, trying to duplicate the task with some some nitty gritty conditionals around RH distro specifics then.

Reply:
```
Syntax Error while loading YAML.
  did not find expected '-' indicator

The error appears to be in '/tmp/cirrus-ci-build/ovn-fake-multinode-utils/playbooks/install-dependencies.yml': line 13, column 9, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

        - containers-common
        state: present
        ^ here
```

I it was the same error before.  I guess it should be:
```
 hosts: ovn_hosts
  tasks:
    - name: Install required packages
      ansible.builtin.package:
        name:
          - git
          - gcc
          - openvswitch
          - python3-pyyaml
          - python3-virtualenv
          - python3-devel
          - containers-common
        skip_broken: true
        state: present
      when: ansible_os_family == "RedHat"
```

Reply:
On the [original change attempt](https://cirrus-ci.com/task/6206298660798464) it specifically choked on the `skip_broken` key

Reply:
Yes, because you didn't have a `name:` qualifier.  So, it tried to interpret `skip_broken` as part of the list.  If we have a `name:` and package list is indented, it shouldn't fail this way, AFAICT.

Reply:
Ha, indeed, good spot, thx!

Reply:
Unfortunately 'skip-broken' is broken (pun intended) on some ansible versions:
https://github.com/ansible/ansible/issues/73072

Officially this should've been starting with ansible 2.14.3 (https://github.com/ansible/ansible/commit/6bcb494f8306615f2b8741dad23529fdcd94626c) but I didn't manage to make it work on our RHEL8.7.  I'm using 

```
# ansible --version
ansible [core 2.14.3]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.9/site-packages/ansible
  ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.9.13 (main, Nov  9 2022, 13:16:24) [GCC 8.5.0 20210514 (Red Hat 8.5.0-15)] (/usr/bin/python3.9)
  jinja version = 3.1.2
  libyaml = True
```

And still getting errors for missing packages:
```
TASK [Install required packages] *************************************************************************************************************************************************************************************************************
fatal: [advnetlab29.anl.eng.rdu2.redhat.com]: FAILED! => {"changed": false, "failures": ["No package openvswitch available."], "msg": "Failed to install some of the specified packages", "rc": 1, "results": []}                            
fatal: [advnetlab11.anl.eng.rdu2.redhat.com]: FAILED! => {"changed": false, "failures": ["No package openvswitch available."], "msg": "Failed to install some of the specified packages", "rc": 1, "results": []}
```

What if we use the following work around for now?

```
diff --git a/ovn-fake-multinode-utils/playbooks/install-dependencies.yml b/ovn-fake-multinode-utils/playbooks/install-dependencies.yml                                                                                                       
index 5229252..608e21b 100644
--- a/ovn-fake-multinode-utils/playbooks/install-dependencies.yml
+++ b/ovn-fake-multinode-utils/playbooks/install-dependencies.yml
@@ -6,15 +6,24 @@
         name:
           - git
           - gcc
-          - openvswitch
           - python3-pyyaml
           - python3-virtualenv
           - python3-devel
           - containers-common
-        skip_broken: true
         state: present
       when: ansible_os_family == "RedHat"

+    # 'skip-broken' might not work with specific ansible versions
+    # so ignore_errors for packages that might not be there.
+    # https://github.com/ansible/ansible/commit/6bcb494f8306
+    - name: Install optional packages
+      ansible.builtin.package:
+        name:
+          - openvswitch
+        state: present
+      ignore_errors: true
+      when: ansible_os_family == "RedHat"
+
     - name: Install required packages
       ansible.builtin.package:
         name:
```

Reply:
Thx for testing @dceara, works for me, I'll include it in the PR